File: code/game/entity/actor/actor.cpp
#include "actor.hpp"

namespace rebubox{

actor::actor(focused_draw_target_t fdtarget, const math::point2d &pos):
    draw_target(fdtarget),
    position(pos)
{}

actor::~actor(){}

math::point2d &actor::get_position(){
    return position;
}

const math::point2d &actor::get_position() const{
    return position;
}

void actor::set_position(const math::point2d &pos){
    position = pos;
}

void actor::update(){}

void actor::draw(){}

focused_draw_target_t actor::get_draw_target(){
    return draw_target;
}

draw_target_view_t actor::get_draw_target() const{
    return draw_target;
}

}

File: scripts/core/actors/actor.lua
local actorsModule = {}

actorsModule.Actor = {
  x = 0, 
  y = 0
}

function actorsModule.Actor:new(object)
  object = object or {}
  setmetatable(object, self)
  self.__index = self
  return object
end

function actorsModule.Actor:update()
end

function actorsModule.Actor:draw()
end

function actorsModule.Actor:getPosition()
  return self.x, self.y
end

function actorsModule.Actor:setPositon(x, y)
  self.x, self.y = x, y
end

return actorsModule